mautic/mautic
# Fix replace trackable tokens from longest to short

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
If URL in the content uses uppercase, then sorting trackable urls from longest to shorten not working correctly. Then the shortened version is replaced first and it makes issues. 



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Create an email and add two links 
- www.mautic.org
- www.MAUTIC.org?test=1

3. Send an email
4. Before www.MAUTIC.org/test was replaced with mautic.tld/r/hashxxx?test=1 (or something weird) and redirect does not work properly
5. After PR works as expected
